activesupport/lib/active_support/core_ext/object/acts_like.rb

```diff
@@ -5,17 +5,19 @@ class Object
   # to define an <tt>acts_like_date?</tt> method, and extends Time to define
   # <tt>acts_like_time?</tt>. As a result, we can do <tt>x.acts_like?(:time)</tt> and
   # <tt>x.acts_like?(:date)</tt> to do duck-type-safe comparisons, since classes that
-  # we want to act like Time simply need to define an <tt>acts_like_time?</tt> method.
+  # we want to act like Time simply need to define an <tt>acts_like_time?</tt>
+  # method that returns true.
   def acts_like?(duck)
     case duck
     when :time
-      respond_to? :acts_like_time?
+      respond_to?(:acts_like_time?) && acts_like_time?
     when :date
-      respond_to? :acts_like_date?
+      respond_to?(:acts_like_date?) && acts_like_date?
     when :string
-      respond_to? :acts_like_string?
+      respond_to?(:acts_like_string?) && acts_like_string?
     else
-      respond_to? :"acts_like_#{duck}?"
+      acts_like_method = :"acts_like_#{duck}?"
+      respond_to?(acts_like_method) && send(acts_like_method)
```

Comment:
What if we did this? 

```suggestion
      respond_to?(acts_like_method) && send(acts_like_method) != false
```

Then the only change would be if someone override the acts like method to explicitly return false. At which point, the current behaviour is surprising, as @mvastola argued in https://github.com/rails/rails/issues/42292

Reply:
I can get on board with that. This also creates an easy outlet to add in a deprecation warning eventually if that is decided (for cases where the value is neither `false` nor truthy).

Reply:
Another option is always to add a config option and -- if the config option isn't explicitly set and the value is neither `true` nor `false` -- show a deprecation warning about the ambiguity while retaining the current functionality for now.

Reply:
> Another option is always to add a config option and -- if the config option isn't explicitly set and the value is neither true nor false -- show a deprecation warning about the ambiguity while retaining the current functionality for now.

I think that is the best option long term. @SkipKayhil do you want to work on this?

Reply:
🦆